src/utils/matter-parser.js

```diff
@@ -0,0 +1,98 @@
+const yaml = require('yaml');
+const fsp = require('fs').promises;
+const glob = require('glob');
+const { promisify } = require('util');
+
+/**
+ * Class for parsing a markdown file into frontmatter and content
+ */
+class MatterParser {
+  /**
+   * Creates a markdown file instance
+   * @param {string} fileName the file name
+   * @param {object} metaData the parsed frontmatter
+   * @param {string} content the content body
+   * @param {string} eol the line-ending style used (\n or \r\n)
+   */
+  constructor(fileName, metaData, content, eol = '\n') {
```

Comment:
Would it be interesting to have the `eol` argument default to `os.EOL` maybe? 
https://nodejs.org/api/os.html#os_os_eol

Reply:
good point :+1: was checking if some existing tools like [markdownlint](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md) have a preference regarding line endings, it doesn't seem to matter. 

I think it's best practice to default to os.EOL in the constructor parameter.